app/models/mentee_application_validation.rb

```diff
@@ -0,0 +1,74 @@
+class MenteeApplicationValidation
+  EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
+
+  def initialize(params, step:)
+    @params = params
+    @step = step
+  end
+
+  def valid?
+    validate.success?
+  end
+
+  def errors
+    validate.messages(full: true)
+  end
+
+  def attrs
+    validate.output
+  end
+
+  private
+
+  def params
```

Comment:
you can add `attr_reader` for both `params` and `step` instead of adding those methods :)